programmers2/모의고사.js

```diff
@@ -0,0 +1,19 @@
+function solution(answers) {
+    const a = [1,2,3,4,5];
+    const b = [2,1,2,3,2,4,2,5];
+    const c = [3,3,1,1,2,2,4,4,5,5];
+    let cnt = [0, 0, 0];
+    
+    for (let i = 0; i < answers.length; i++) {
+        if (a[i % a.length] === answers[i]) cnt[0]++;
+        if (b[i % b.length] === answers[i]) cnt[1]++;
+        if (c[i % c.length] === answers[i]) cnt[2]++;
+    };
+    
+    const max = Math.max(...cnt);
+    let result = [];
```

Comment:
배열 같은 경우는 iterable 객체라고 합니다.
iterable 객체는 배열, 오브젝트 등의 타입들을 통틀어 이야기 하는데요.
즉, 내부 원소들이 존재하는 객체들을 iterable 객체 라고 합니다.
Array, Object, Set, Map 등이 이에 해당하겠네요.

iterable 객체들의 특징은 mutable 하다는 것인데요.
mutable의 사전적 정의는 '변이 가능한' 입니다.
즉, 내부 원소들이 변할 수 있다는 것이지요.
따라서 iterable 객체들은 무조건 const로 선언하는 것이 관례입니다.
특수한 경우에만 let을 선언해주고 있어요.

예를 들어, 내부 원소를 바꾸는 것이 아니라 배열을 통째로 변경해야하는 경우에는 let을 선언해줍니다. 아래처럼요.
```js
let arr = [1, 2, 3];
arr = ["1", "2", "3"];
```
하지만, 웬만해서는 위처럼 배열 전체를 바꾸는 행위는 지양해야 합니다.
협업할 때 누군가의 실수로 값을 바꿀 수도 있기 때문이죠.
이와 같은 이유로 let보다는 const를 사용하는 것입니다.

```suggestion
    const result = [];
```

Reply:
아아 저 내용은 처음 알았네요 !  더 찾아서 공부해 볼게욤... 감사합니당